### DISCUSSION TITLE
pyarmor.cli.core error on registration

### DISCUSSION THREAD

**Discussion by Serob** [dashingsoft/pyarmor#1056] [Q&A]
I have purchased a registration at 2019 and was using it with version 5.8.9 successfully.
Now I decided to upgrade my python and pyarmor versions to the latest ones, and when try to register the file by calling `pyarmor reg path-to-pyarmor-regfile-1.zip`, it gives an unknown error

![image](https://user-images.githubusercontent.com/10531198/228867063-f687e6f8-9293-4198-9838-0cd69b672307.png)

How can I use the old license for the newest (8.0.22) version?

**Comment by jondy** [comment]
Please check project [README.md](https://github.com/dashingsoft/pyarmor)

**Reply by Serob** [reply]
In licence migration [section](https://pyarmor.readthedocs.io/en/stable/licenses.html#freely-to-pyarmor-basic) it is said:

```
First find the activation file pyarmor-regcode-xxxx.txt, which is sent to registration email when purchasing the license.
```

But when I bought the license it was in .zip format, not .txt. Where can I find the .txt then?

**Reply by jondy** [reply]
You should receive `.txt` except you purchase multiple licenses in one order, generally its name is `multiple-licenses.zip`.

**Reply by Serob** [reply]
When I bought it I received only `pyarmor-regfile-1.zip` file, which includes 3 files:  _capsule, license, README_. The provided URL is inactive now:
![image](https://user-images.githubusercontent.com/10531198/229107187-555b933a-ba62-4ec3-b42d-7679ebad5d3f.png)
How/when should I receive `.txt` ?

**Comment by jondy** [comment]
In this case, this code can't be upgraded freely.

**Reply by Serob** [reply]
What's the possible approach? Buy new license?

**Reply by jondy** [reply]
Yes. If using Pyarmor 8.0, it need buy new license.

**Reply by Serob** [reply]
Would pyarmor 7 support python 11?
